code/modules/admin/verbs/modifyvariables.dm

```diff
@@ -1,494 +1,520 @@
-		if(findtext(var_value,"\["))
-			var/process_vars = alert(usr,"\[] detected in string, process as variables?","Process Variables?","Yes","No")
-			if(process_vars == "Yes")
-				var/list/varsvars = string2listofvars(var_value, O)
-				for(var/V in varsvars)
-					var_value = replacetext(var_value,"\[[V]]","[O.vars[V]]")
 
-	return var_value
+		if (VV_ATOM_REFERENCE)
+			var/type = pick_closest_path(FALSE)
+			var/subtypes = vv_subtype_prompt(type)
+			if (subtypes == null)
+				.["class"] = null
+				return
+			var/list/things = vv_reference_list(type, subtypes)
+			var/value = input("Select reference:", "Reference", current_value) as null|anything in things
+			if (!value)
+				.["class"] = null
+				return
+			.["value"] = things[value]
 
+		if (VV_DATUM_REFERENCE)
+			var/type = pick_closest_path(FALSE, get_fancy_list_of_datum_types())
+			var/subtypes = vv_subtype_prompt(type)
+			if (subtypes == null)
+				.["class"] = null
+				return
+			var/list/things = vv_reference_list(type, subtypes)
+			var/value = input("Select reference:", "Reference", current_value) as null|anything in things
+			if (!value)
+				.["class"] = null
+				return
+			.["value"] = things[value]
 
-/client/proc/mod_list_add(list/L, atom/O, original_name, objectvar)
+		if (VV_MOB_REFERENCE)
+			var/type = pick_closest_path(FALSE, make_types_fancy(typesof(/mob)))
+			var/subtypes = vv_subtype_prompt(type)
+			if (subtypes == null)
+				.["class"] = null
+				return
+			var/list/things = vv_reference_list(type, subtypes)
+			var/value = input("Select reference:", "Reference", current_value) as null|anything in things
+			if (!value)
+				.["class"] = null
+				return
+			.["value"] = things[value]
 
-	var/class = "text"
-	if(src.holder && src.holder.marked_datum)
-		class = input("What kind of variable?","Variable Type") as null|anything in list("text",
-			"num","type","reference","mob reference", "icon","file","list","edit referenced object","restore to default", "new atom", "new datum","marked datum ([holder.marked_datum.type])")
-	else
-		class = input("What kind of variable?","Variable Type") as null|anything in list("text",
-			"num","type","reference","mob reference", "icon","file","list","edit referenced object","restore to default", "new atom", "new datum")
 
-	if(!class)
-		return
 
-	if(holder.marked_datum && class == "marked datum ([holder.marked_datum.type])")
-		class = "marked datum"
+		if (VV_CLIENT)
+			.["value"] = input("Select reference:", "Reference", current_value) as null|anything in clients
+			if (.["value"] == null)
+				.["class"] = null
+				return
 
-	var/var_value = null
 
-	switch(class)
+		if (VV_FILE)
+			.["value"] = input("Pick file:", "File") as null|file
+			if (.["value"] == null)
+				.["class"] = null
+				return
 
-		if("text")
-			var_value = input("Enter new text:","Text") as message
 
-		if("num")
-			var_value = input("Enter new number:","Num") as num
+		if (VV_ICON)
+			.["value"] = input("Pick icon:", "Icon") as null|icon

```

Comment:
Would be nice if icon_state was populated from icon_states() but that can wait for another PR